src/providers/AuthenticationProvider.tsx

```diff
@@ -1,11 +1,82 @@
-import React, { createContext, FC, useContext, useState } from "react";
+/* eslint-disable no-console */
+import { googleLogout, useGoogleLogin } from "@react-oauth/google";
+import axios from "axios";
+import React, { createContext, FC, useContext, useEffect, useState } from "react";
 
-// =========== INIT FUNCTION =========== Will initialize the provider state for us - this lets us ifer the type of the return and ensure type safety
+interface GoogleUser {
+    access_token: string;
+    authuser: string;
+    expires_at: number;
+    prompt: string;
+    scope: string;
+    token_type: string;
+}
+interface GoogleProfile {
+    email: string;
+    family_name: string;
+    given_name: string;
+    id: string;
+    locale: string;
+    name: string;
+    picture: string;
+    verified_email: boolean;
+}
+
+// =========== INIT FUNCTION =========== Will initialize the provider state for us - this lets us infer the type of the return and ensure type safety
 const createAuthenticationProviderState = () => {
-    const [user, setUser] = useState([]);
-    const [profile, setProfile] = useState([]);
+    const [user, setUser] = useState<GoogleUser>();
+    const [profile, setProfile] = useState<GoogleProfile>();
```

Comment:
What’s the difference between `user` and `profile`, when does each get set?

Reply:
Yelloow so user is an object with the basic info of the users account, in this case access_token being one of them, which is used for the axios http request. The profile object has more information that we can use in the app, such as the values seen in the interface, but among these picture, name, full name, locale and verified email could be the most pertinent as of right now. As to when do they get set, the setUser is used in the handleSuccess handler that is called within the handleLogin method, that in turn makes use of the useGoogleLogin hook. The setProfile is called within the the axios .then() method, which means that the promise of the request has been fulfilled with the given client access_token and is then given the data from the response of the request. Profile is the only one being used for the data extraction in the app.tsx component while user is just used for the request in this instance.